File: src/main/java/guru/xgm/language/lni/lang/folding/LniCodeFoldingOptionsProvider.java
package guru.xgm.language.lni.lang.folding;

import com.intellij.application.options.editor.CodeFoldingOptionsProvider;
import com.intellij.openapi.options.BeanConfigurable;

public class LniCodeFoldingOptionsProvider extends BeanConfigurable<LniCodeFoldingSettings> implements CodeFoldingOptionsProvider {
    public LniCodeFoldingOptionsProvider() {
        super(LniCodeFoldingSettings.getInstance(), "LNI");
        LniCodeFoldingSettings settings = getInstance();
        assert settings != null;
        checkBox("Item", settings::isFoldItems, settings::setFoldItems);
    }
}

File: src/main/kotlin/guru/xgm/language/lni/lang/folding/LniCodeFoldingOptionsProvider.kt
package guru.xgm.language.lni.lang.folding

import com.intellij.application.options.editor.CodeFoldingOptionsProvider
import com.intellij.openapi.options.BeanConfigurable
import guru.xgm.language.lni.lang.folding.LniCodeFoldingSettings.Companion.instance

class LniCodeFoldingOptionsProvider : BeanConfigurable<LniCodeFoldingSettings?>(instance, "LNI"),
    CodeFoldingOptionsProvider {
    init {
        checkBox("Item", instance::isFoldItem, instance::isFoldItem.setter)
        checkBox("List", instance::isFoldList, instance::isFoldList.setter)
    }
}
